src/components/v5/frame/ColonyHome/partials/ReputationChart/hooks/useContributorsByDomain.tsx

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react';
+
+import { useMemberContext } from '~context/MemberContext/MemberContext.ts';
+import { type ColonyContributorFragment } from '~gql';
+import useGetSelectedDomainFilter from '~hooks/useGetSelectedDomainFilter.tsx';
+
+import { type ContributorItem } from '../types.ts';
+
+export const getContibutorsByDomain = (
+  members: ColonyContributorFragment[],
+  selectedTeamId: number | undefined,
+): ContributorItem[] => {
+  const isAllTeamsSelected = selectedTeamId === undefined;
+
+  return members.map((contributor) => {
+    const { contributorAddress, colonyReputationPercentage, user, reputation } =
+      contributor;
+
+    const teamReputationPercentage = reputation?.items?.find(
+      (item) => item?.domain?.nativeId === selectedTeamId,
+    )?.reputationPercentage;
+
+    return {
+      user,
+      walletAddress: contributorAddress,
+      reputation: isAllTeamsSelected
+        ? colonyReputationPercentage
+        : teamReputationPercentage,
+    };
+  });
+};
+
+export const useContributorsByDomain = () => {
+  const { totalMembers, loading } = useMemberContext();
+
+  const allContributors = useMemo(
+    () =>
+      totalMembers.filter(({ hasReputation }) => hasReputation).sort() ?? [],
+    [totalMembers],
+  );
+
+  const selectedDomain = useGetSelectedDomainFilter();
+
+  const contributorsWithReputationByDomain = useMemo(
+    () => getContibutorsByDomain(allContributors, selectedDomain?.nativeId),
+    [allContributors, selectedDomain],
+  );
+
+  const contributorsList = contributorsWithReputationByDomain
+    .filter((contributor) => !!contributor.reputation)
+    .sort((a, b) => Number(b.reputation) - Number(a.reputation));
```

Comment:
I'm sure this has been tested but just wanted to confirm if we don't want to add any sort of null check for the reputation field values here just in case? Looking at the code, it can be `undefined` and `Number(undefined)` is `NaN`